thamos/lib.py

```diff
@@ -97,7 +101,30 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-def _wait_for_analysis(status_func: Callable[..., Any], analysis_id: str) -> None:
+def get_analysis_status(analysis_id: str, source: str) -> bool:
```

Comment:
I think it would be better to rename this function - it's not obtaining status, but is asking if the analysis is ready - `is_analysis_ready`?